sematic/db/models/mixins/json_encodable_mixin.py

```diff
@@ -26,25 +27,26 @@ def to_json_encodable(self, redact: bool = True):
     @classmethod
     def from_json_encodable(cls, json_encodable):
         field_dict = {
-            column.key: cls.field_from_json_encodable(column.key, json_encodable)
+            column.key: cls.field_from_json_encodable(column, json_encodable)
             for column in inspect(cls).attrs
             if column.key in json_encodable
         }
         return cls(**field_dict)
 
     @classmethod
-    def field_from_json_encodable(cls, field_name, json_encodable):
-        return _from_json_encodable(
-            json_encodable[field_name], getattr(cls, field_name)
-        )
+    def field_from_json_encodable(cls, column: Column, json_encodable):
+        field_name = column.info.get(ALIAS_KEY, column.key)
+
+        return _from_json_encodable(json_encodable[field_name], column)
```

Comment:
The code as written says the alias should take precedence over the current actual column name. I think a more desirable behavior would be this:

```suggestion
    def field_from_json_encodable(cls, column: Column, json_encodable):
        field_name = column.key
        field_alias = column.info.get(
            ALIAS_KEY,
            column.key,
        )
        if (
            field_name not in json_encodable and
            field_alias in json_encodable
        ):
            field_name = field_alias
           
        return _from_json_encodable(
            json_encodable[field_name],
            column,
        )
```

Some examples to illustrate desired behavior:

```
# column has key "foo" and alias "foo_old":
>> T.field_from_json_encodable(column, {"foo": "bar"}) # A
"bar"
>> T.field_from_json_encodable(column, {"foo_old": "bar"}) # B
"bar"
>> T.field_from_json_encodable(column, {"baz": "qux"}) # C
KeyError: Missing key "foo"
>> T.field_from_json_encodable(column, {"foo": "bar", {"foo_old": "bar_old"}}) # D
"bar"
```

I believe my suggestion would give this behavior, but yours would error for `A` and give a key error listing "foo_old" for `C`. `D` probably shouldn't happen, but ideally we'd use the current column name's value if it ever does.